ci/management/commands/generate_reports.py

```diff
@@ -303,3 +354,54 @@ def handle_health_worker_registrations(
                 'Cadre': operator_details.get('role'),
                 'Number of Registrations': count,
             })
+
+    def handle_enrollments(self, sheet, sbm_client, ids_client, start_date,
+                           end_date):
+
+        sheet.set_header([
+            'Message set',
+            'Roleplayer',
+            'Total enrolled',
+            'Enrolled in period',
+            'Enrolled and opted out in period',
+            'Enrolled and completed in period',
+        ])
+
+        subscriptions = self.get_subscriptions(
+            sbm_client,
+            created_before=end_date.isoformat())
+
+        data = collections.defaultdict(partial(collections.defaultdict, int))
+        for subscription in subscriptions:
+            messageset = self.get_messageset(
+                            sbm_client, subscription['messageset'])
+            identity = self.get_identity(ids_client, subscription['identity'])
+
+            messageset_name = messageset['short_name'].split('.')[0]
+
+            receiver_role = 'None'
+            if identity:
+                receiver_role = identity.get('details', {}).get(
+                                    'receiver_role', 'None')
+
+            data[messageset_name, receiver_role]['total'] += 1
+
+            if parse_datetime(subscription['created_at']) > start_date:
+                data[messageset_name, receiver_role]['total_period'] += 1
+
+                if (not subscription['active'] and
+                        not subscription['completed']):
+                    data[messageset_name, receiver_role]['optouts'] += 1
+
+                if subscription['completed']:
+                    data[messageset_name, receiver_role]['completed'] += 1
+
+        for key in sorted(data.keys()):
```

Comment:
Possibly easier to read if we do `for (messageset_name, receiver_role), counts in data.items():`, but either way it's fine.